app/javascript/modules/listingDetails/ListingDetailsEligibility.tsx

```diff
@@ -9,19 +9,74 @@ import {
 } from "@bloom-housing/ui-components"
 import { RailsListing } from "../listings/SharedHelpers"
 import { isHabitatListing, isSale } from "../../util/listingUtil"
-import { renderMarkup } from "../../util/languageUtil"
+import { renderMarkup, defaultIfNotTranslated } from "../../util/languageUtil"
 import { BeforeApplyingForSale, BeforeApplyingType } from "../../components/BeforeApplyingForSale"
 import { ListingDetailsPreferences } from "./ListingDetailsPreferences"
+import RailsUnit from "../../api/types/rails/listings/RailsUnit"
 
 export interface ListingDetailsEligibilityProps {
   listing: RailsListing
   imageSrc: string
 }
 
+export interface ReducedUnit {
+  name: string
+  numberOfUnits: number
+}
+
 export const ListingDetailsEligibility = ({
   listing,
   imageSrc,
 }: ListingDetailsEligibilityProps) => {
+  const priorityUnits = []
+
+  listing.Units.forEach((unit: RailsUnit) => {
+    const priorityUnit = priorityUnits.find((priorityUnit: ReducedUnit) => {
+      return priorityUnit.name === unit.Priority_Type
+    })
+
+    if (unit.Priority_Type && !priorityUnit) {
+      priorityUnits.push({
+        name: unit.Priority_Type,
+        numberOfUnits: 1,
+      })
+    }
+
+    if (unit.Priority_Type && priorityUnit) {
+      priorityUnit.numberOfUnits++
+    }
+  })
+
+  const priorityLabelMap = {
+    "Mobility impairments": {
+      titleTranslation: "listings.prioritiesDescriptor.mobility",
+      descriptionTranslation: "listings.unitsHaveAccessibilityFeaturesFor.impaired mobility",
+    },
+    "Hearing/Vision impairments": {
+      titleTranslation: "listings.prioritiesDescriptor.hearingVision",
+      descriptionTranslation:
+        "listings.unitsHaveAccessibilityFeaturesFor.impaired vision and/or hearing",
+    },
+    "Hearing impairments": {
+      titleTranslation: "listings.prioritiesDescriptor.hearing",
+      descriptionTranslation: "listings.unitsHaveAccessibilityFeaturesFor.impaired hearing",
+    },
+    "Mobility/Hearing/Vision impairments": {
+      titleTranslation: "listings.prioritiesDescriptor.mobilityHearingVision",
+      descriptionTranslation:
+        "listings.unitsHaveAccessibilityFeaturesFor.impaired mobility, hearing and/or vision",
+    },
+    "Vision impairments": {
+      titleTranslation: "listings.prioritiesDescriptor.vision",
+      descriptionTranslation: "listings.unitsHaveAccessibilityFeaturesFor.impaired vision",
+    },
+    "Hearing/Vision (Communication)": {
+      titleTranslation: "listings.prioritiesDescriptor.hearingVisionCommunication",
+      descriptionTranslation:
+        "listings.unitsHaveAccessibilityFeaturesFor.impaired hearing and/or vision (communication)",
+    },
+  }
```

Comment:
is this data structure needed? could we change the text prop names to include the listing value, similar to what was done [here](https://github.com/SFDigitalServices/sf-dahlia-web/blob/69c740b8e1fd180e9fecd1dae6391940e085db6b/app/javascript/modules/listingDetailsLottery/ListingDetailsLotteryPreferences.tsx#L35-L35)?

then you could access the values with `<prefix>.${priorityUnit.name}.title` and `<prefix>.${priorityUnit.name}.description`?